app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -169,74 +170,176 @@ function userOption(userOptionKey) {
+
+  _updateStatus(statuses) {
+    if (statuses.hasOwnProperty(this.id)) {
+      this.set("status", statuses[this.id]);
+    }
+  },
+});
+
+export default class User extends RestModel.extend(
+  Evented,
+  UserStatusTracking
+) {
+  @service appEvents;
+
+  @service userTips;
+
+  @userOption("mailing_list_mode") mailing_list_mode;
+  @userOption("external_links_in_new_tab") external_links_in_new_tab;
+  @userOption("enable_quoting") enable_quoting;
+  @userOption("dynamic_favicon") dynamic_favicon;
+  @userOption("automatically_unpin_topics") automatically_unpin_topics;
+  @userOption("likes_notifications_disabled") likes_notifications_disabled;
+  @userOption("hide_profile_and_presence") hide_profile_and_presence;
+  @userOption("title_count_mode") title_count_mode;
+  @userOption("enable_defer") enable_defer;
+  @userOption("timezone") timezone;
+  @userOption("skip_new_user_tips") skip_new_user_tips;
+  @userOption("default_calendar") default_calendar;
+  @userOption("bookmark_auto_delete_preference")
+  bookmark_auto_delete_preference;
+  @userOption("seen_popups") seen_popups;
+  @userOption("should_be_redirected_to_top") should_be_redirected_to_top;
+  @userOption("redirected_to_top") redirected_to_top;
+  @userOption("treat_as_new_topic_start_date") treat_as_new_topic_start_date;
+
+  @gt("private_messages_stats.all", 0) hasPMs;
+  @gt("private_messages_stats.mine", 0) hasStartedPMs;
+  @gt("private_messages_stats.unread", 0) hasUnreadPMs;
+  @url("id", "username_lower", "/admin/users/%@1/%@2") adminPath;
+  @equal("trust_level", 0) isBasic;
+  @equal("trust_level", 3) isRegular;
+  @equal("trust_level", 4) isLeader;
+  @or("staff", "isLeader") canManageTopic;
+  @alias("sidebar_category_ids") sidebarCategoryIds;
+  @alias("sidebar_sections") sidebarSections;
+  @mapBy("sidebarTags", "name") sidebarTagNames;
+  @filterBy("groups", "has_messages", true) groupsWithMessages;
+
+  numGroupsToDisplay = 2;
 
   @discourseComputed("can_be_deleted", "post_count")
   canBeDeleted(canBeDeleted, postCount) {
     const maxPostCount = this.siteSettings.delete_all_posts_max;
     return canBeDeleted && postCount <= maxPostCount;
-  },
+  }
 
   @discourseComputed()
   stream() {
     return UserStream.create({ user: this });
-  },
+  }
 
   @discourseComputed()
   bookmarks() {
     return Bookmark.create({ user: this });
-  },
+  }
 
   @discourseComputed()
   postsStream() {
     return UserPostsStream.create({ user: this });
-  },
+  }
 
   @discourseComputed()
   userDraftsStream() {
     return UserDraftsStream.create({ user: this });
-  },
+  }
 
-  staff: computed("admin", "moderator", {
-    get() {
-      return this.admin || this.moderator;
-    },
+  @computed("admin", "moderator")
+  get staff() {
+    return this.admin || this.moderator;
+  }
 
-    // prevents staff property to be overridden
-    set() {
-      return this.admin || this.moderator;
-    },
-  }),
+  // prevents staff property to be overridden
+  set staff(value) {}
```

Comment:
This might not be needed in a native class?

```js
a = { get x() { return 1; } }
// {}
a.x
// 1
a.x = 3
// 3
a.x
// 1
```

```suggestion
```

Reply:
Ember's `set()` method throws a spanner in the works:

<img width="578" alt="SCR-20240209-ppfe" src="https://github.com/discourse/discourse/assets/6270921/f6e40976-5f5a-4aa8-b9d6-af97da9306b4">

So I think this no-op setter is needed for now.

Of course, ideally nobody would be trying to set this property in the first place... but fixing all those places is more work than I'd like to do for this PR.